security/go-mod-and-go-sum.md

```diff
@@ -0,0 +1,135 @@
+# Go.mod and go.sum
+
+This investigatio clarifies the relation between `go.mod` and `go.sum` files and
+their respective usages.
+
+## Some basic definitions
+
+First, let's do some research what the files are supposed to be doing.
+
+### Go.mod
+
+- "Each Go module is defined by a go.mod file that describes the module’s
+   properties, including its dependencies on other modules and on versions of
+   Go." - [go.dev](https://go.dev/doc/modules/gomod-ref),
+   [go.dev](https://go.dev/ref/mod)
+
+- "The go.mod file lists the specific versions of the dependencies that your
+   project uses." -
+   [FreeCodeCamp](https://www.freecodecamp.org/news/golang-environment-gopath-vs-go-mod/#:~:text=sum%20file.-,The%20go.,all%20projects%20on%20your%20system.)
+
+- "No, go.sum is not a lock file. The go.mod files in a build provide
```

Comment:
The same line is added in go.sum section as well, intentional ?

Reply:
Yes, it talks about both in a meaningful way. For go.mod it talks about reproducability and for go.sum it talks about not being a lock file, and is then followed up by a longer snippet.